From 864e445974de7c843cdb4ee008bad0a550e5af6e Mon Sep 17 00:00:00 2001
From: Jiří Techet <techet@gmail.com>
Date: Thu, 06 Oct 2011 22:06:59 +0000
Subject: When re-creating scale, remove all actors first

Otherwise the previously inserted actors are still used for redrawing.
---
diff --git a/champlain/champlain-scale.c b/champlain/champlain-scale.c
index 34f5f06..a4befcd 100644
--- a/champlain/champlain-scale.c
+++ b/champlain/champlain-scale.c
@@ -476,6 +476,8 @@ create_scale (ChamplainScale *scale)
   gfloat width;
   ChamplainScalePrivate *priv = scale->priv;
 
+  clutter_group_remove_all (priv->content_group);
+  
   text = clutter_text_new_with_text ("Sans 9", "X km");
   clutter_actor_set_name (text, "scale-far-label");
   clutter_container_add_actor (CLUTTER_CONTAINER (priv->content_group), text);
--
cgit v0.9.0.2
